### PR TITLE
migrate from dep to Go modules

### PR DESCRIPTION
**Pull request by xh3b4sd**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	github.com/giantswarm/e2eclients v0.2.0
+	github.com/giantswarm/e2esetup v0.1.0
+	github.com/giantswarm/e2etemplates v0.2.0
+	github.com/giantswarm/e2etests v0.1.0
+	github.com/giantswarm/errors v0.2.0
+	github.com/giantswarm/exporterkit v0.2.0
+	github.com/giantswarm/helmclient v0.2.0
+	github.com/giantswarm/ipam v0.2.0
+	github.com/giantswarm/k8sclient v0.2.0
+	github.com/giantswarm/k8scloudconfig v0.0.0-20200325205340-f571257e51f7
```
Comment (xh3b4sd) [inline go.mod]: Well, this is complete bullshit. You can't define `v6.0.0`. Don't know right now how the go folks thought this should be.

Comment (xh3b4sd) [reply go.mod]: Sorted.

**Comment by xh3b4sd** [comment]
Well, the `Dockerfile` wants to add `k8scloudconfig` to the container based on the `vendor` directory, which is gone with go modules. How should that work now? Build fails. /cc @corest @kopiczko 

```
Step 6/9 : ADD vendor/github.com/giantswarm/k8scloudconfig/ /opt/ignition
ADD failed: stat /var/lib/docker/tmp/docker-builder553670661/vendor/github.com/giantswarm/k8scloudconfig: no such file or directory
```

**Comment by wtty-fool** [comment]
@xh3b4sd Maybe we can temporarily vendor dependencies using `go mod vendor`? Just for this step.

**Comment by wtty-fool** [comment]
@giantswarm/wg-go-modules Ready for your reviews, gentlemen.

**Comment by wtty-fool** [comment]
Holding off merging this until I test it one last time to be 100% percent sure :sweat_smile:

**Comment by MarcelMue** [comment]
@wtty-fool Can you give a rough estimate here? We are right now doing some stuff on k8scc and would be cool to use this.

**Comment by wtty-fool** [comment]
@MarcelMue today, just waiting for _gauss/ginger_ to become available.

**Comment by wtty-fool** [comment]
Tested on _gauss_.

**Pull request merged**